bigquery/google/cloud/bigquery/table.py

```diff
@@ -48,6 +48,42 @@
 _DEFAULT_NUM_RETRIES = 6
 
 
+class TableReference(object):
+    """TableReferences are pointers to tables.
+
+    See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables
+
+    :type dataset_ref: :class:`google.cloud.bigquery.dataset.DatasetReference`
+    :param dataset_ref: a pointer to the dataset
+
+    :type table_id: str
+    :param table_id: the ID of the table
+    """
+
+    def __init__(self, dataset_ref, table_id):
+        self._dataset_ref = dataset_ref
+        self._table_id = table_id
+
+    @property
+    def dataset_ref(self):
```

Comment:
Should this be called `dataset`, for consistency with `client.dataset`?

Reply:
We are documenting it as a DatasetReference, so it wouldn't be as likely to be confused with the Dataset class, which was my original fear.